packages/api/endpoints/elasticsearch.js

```diff
@@ -0,0 +1,186 @@
+
+  if (destExists) {
+    return res.boom.badRequest(`Destination index ${destIndex} exists. Please specify an index name that does not exist.`);
+  }
+
+  // create destination index
+  await esClient.indices.create({
+    index: destIndex,
+    body: { mappings }
+  });
+
+  log.info(`Created destination index ${destIndex}.`);
+
+  // reindex
+  esClient.reindex({
+    body: {
+      source: { index: sourceIndex },
+      dest: { index: destIndex }
+    }
+  });
+
+  const message = `Reindexing to ${destIndex} from ${sourceIndex}. Check the reindex-status endpoint for status.`;
+
+  return res.status(200).send({ message });
+}
+
+async function reindexStatus(req, res) {
+  const esClient = await Search.es();
+
+  const reindexTaskStatus = await esClient.tasks.list({ actions: ['*reindex'] });
+
+  await esClient.indices.refresh();
+
+  const indexStatus = await esClient.indices.stats({
+    metric: 'docs'
+  });
+
+  const status = {
+    reindexStatus: reindexTaskStatus,
+    indexStatus
+  };
+
+  return res.send(status);
+}
+
+async function completeReindex(req, res) {
+  const deleteSource = req.body.deleteSource;
+  const aliasName = req.body.aliasName || defaultIndexAlias;
+  const sourceIndex = req.body.sourceIndex;
+  const destIndex = req.body.destIndex;
+
+  const esClient = await Search.es();
+
+  if (!sourceIndex || !destIndex) {
+    return res.boom.badRequest('Please explicity specify a source and destination index.');
+  }
+
+  if (sourceIndex === destIndex) {
+    return res.boom.badRequest('The source index cannot be the same as the destination index.');
+  }
+
+  const sourceExists = await esClient.indices.exists({ index: sourceIndex });
+
+  if (!sourceExists) {
+    return res.boom.badRequest(`Source index ${sourceIndex} does not exist.`);
+  }
+
+  const destExists = await esClient.indices.exists({ index: destIndex });
+
+  if (!destExists) {
+    return res.boom.badRequest(`Destination index ${destIndex} does not exist.`);
+  }
+
+  await esClient.indices.updateAliases({
+    body: {
+      actions: [
+        { remove: { index: sourceIndex, alias: aliasName } },
+        { add: { index: destIndex, alias: aliasName } }
+      ]
+    }
+  }).then(() => {
+    log.info(`Removed alias ${aliasName} from index ${sourceIndex} and added alias to ${destIndex}`);
+  }, (err) =>
+    res.boom.badRequest(`Error removing alias ${aliasName} from index ${sourceIndex} and adding alias to ${destIndex}: ${err}`));
+
+  let message = `Reindex success - alias ${aliasName} now pointing to ${destIndex}`;
+
+  if (deleteSource) {
+    await esClient.indices.delete({ index: sourceIndex });
+    log.info(`Deleted index ${sourceIndex}`);
+    message = `${message} and index ${sourceIndex} deleted`;
+  }
+
+  return res.send({ message });
+}
+
+// express routes
+router.put('/create-snapshot', createEsSnapshot);
+router.put('/reindex', reindex);
+router.get('/reindex-status', reindexStatus);
+router.put('/complete-reindex', completeReindex);
```

Comment:
I could be wrong, but I think this should be a POST endpoint and not a PUT. The reason why is that PUT endpoints should be idempotent, but this endpoint is not: if you try to submit the same request twice, the second time you'll get a different response because it deleted the source index (assuming the first request succeeded).